### PR TITLE
RBCA: Better separation between action set svc and store

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

Action set code refactor to better separate between the service and the store. Should not have any impact on the behaviour.

**Why do we need this feature?**

Code clean up

**Who is this feature for?**

Internal change

### CODE REVIEW

File: pkg/services/accesscontrol/resourcepermissions/service.go [hunk 1]
```diff
@@ -23,7 +26,7 @@ import (
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-var _ pluginaccesscontrol.ActionSetRegistry = (*InMemoryActionSets)(nil)
+var _ pluginaccesscontrol.ActionSetRegistry = (ActionSetService)(nil)
```
Comment (eleijonmarck) [inline pkg/services/accesscontrol/resourcepermissions/service.go:29]: yea i like this refactoring. the naming makes sense and just separating the svc vs the store.

File: pkg/services/accesscontrol/resourcepermissions/service.go [hunk 2]
```diff
@@ -450,21 +453,126 @@ type ActionSet struct {
...
-	actionSets := &InMemoryActionSets{
-		features:           features,
-		log:                log.New("resourcepermissions.actionsets"),
-		actionSetToActions: make(map[string][]string),
-		actionToActionSets: make(map[string][]string),
+	return &ActionSetSvc{
+		features: features,
+		store:    NewInMemoryActionSetStore(features),
+	}
+}
```
Comment (eleijonmarck) [inline pkg/services/accesscontrol/resourcepermissions/service.go:481]: this is just me but, should we have a logger here as well. that is being used for the store as well.

```suggestion
	return &ActionSetSvc{
		features: features,
		log: log.New("actionsetlogger"),
		store:    NewInMemoryActionSetStore(features),
	}
}
```

**Review by Jguer — APPROVED** [review]

File: pkg/services/accesscontrol/resourcepermissions/service.go (hunk 2 above)
Comment (IevaVasiljeva) [reply pkg/services/accesscontrol/resourcepermissions/service.go:481]: I think we can add it to the service if we start using it in the service, but currently it feels unnecessary imo.

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**